### ISSUE TITLE
How to use moco pretraining model?

### ISSUE THREAD

**Issue by EdyQiu** [open-mmlab/mmdetection#5284]
Hi, I am investigating how pretraining model will affect detection performance, but I could not successfully use any of them.

The setting is 24 epoch, step = [12, 18], learning rate = 0.02, 4GPUs are used.

When using the default torchvision resnet 50, everything is fine.

When using BYOL pretraining model by load_from  =  'https://storage.googleapis.com/deepmind-byol/checkpoints/pretrain_res50x1.pkl', the error showed " No module named 'haiku'" (I think the error  occurred when reading .pkl file), but I failed to find a package called "haiku" using pip or conda.

When using MOCO pretraining model by load_from  =  ''https://dl.fbaipublicfiles.com/moco/moco_checkpoints/moco_v2_800ep/moco_v2_800ep_pretrain.pth.tar'', it showed "unexpected key in source state_dict: encoder_q.conv1 ……………………“  I am really confused, since all I want is a resnet 50 pretrained model, the architecture for all pretrained model should be the same right? The training result is even worse than training from scratch in the end.

The same issue occured when using SWAV pretrained model('https://dl.fbaipublicfiles.com/deepcluster/swav_800ep_pretrain.pth.tar') The error showed "unexpected key in source state_dict: projection_head.0.weight, projection_head.0.bias, projection_head.1.weight ……”  The loss during training is always "nan" with lr = 0.02

Therefore, I am here to ask how should I used a ResNet 50 pretrained model which was pretrained from any self-supervised methods?

**Comment by hhaAndroid** [comment]
Hi, @EdyQiu Thanks for your attention. We have not tried the above approach, but we will look at the cause of the error. I will report the results to you immediately.

**Comment by ZwwWayne** [comment]
This is because the key names in those models are different from mmdet, you need to convert them manually.

**Comment by ZwwWayne** [comment]
@EdyQiu ，
See this PR https://github.com/open-mmlab/mmdetection/pull/5286. MoCov2 has been supported and the loss does not go Nan.

**Comment by EdyQiu** [comment]
> This is because the key names in those models are different from mmdet, you need to convert them manually.

I see, I will read the .pth file to see how these layer names are related. 
After I finish the trial I will close this issue.

Thanks so much for your quick reply!
